homeassistant/components/prosegur/alarm_control_panel.py

```diff
@@ -0,0 +1,94 @@
+"""Support for Prosegur alarm control panels."""
+import logging
+
+from pyprosegur.auth import Auth
+from pyprosegur.installation import Installation, Status
+
+import homeassistant.components.alarm_control_panel as alarm
+from homeassistant.components.alarm_control_panel import (
+    SUPPORT_ALARM_ARM_AWAY,
+    SUPPORT_ALARM_ARM_HOME,
+)
+from homeassistant.const import (
+    STATE_ALARM_ARMED_AWAY,
+    STATE_ALARM_ARMED_HOME,
+    STATE_ALARM_DISARMED,
+)
+
+from . import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the Prosegur alarm control panel platform."""
+    async_add_entities(
+        [ProsegurAlarm(entry.data["contract"], hass.data[DOMAIN][entry.entry_id])]
+    )
+
+
+class ProsegurAlarm(alarm.AlarmControlPanelEntity):
+    """Representation of a Prosegur alarm status."""
+
+    def __init__(self, contract: str, auth: Auth):
+        """Initialize the Prosegur alarm panel."""
+        self._state = None
+        self._changed_by = None
+
+        self._installation = None
+        self.contract = contract
+        self._auth = auth
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return f"contract {self.contract}"
```

Comment:
A lot of these entity property methods can be replaced by their class attributes instead.

See: <https://developers.home-assistant.io/docs/core/entity#entity-class-attributes>